File: backend/src/main/java/edu/cit/mahumot/daycarelog/features/guardians/Guardian.java
package edu.cit.mahumot.daycarelog.features.guardians;

import jakarta.persistence.*;

@Entity
@Table(name = "guardians")
public class Guardian {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "child_id", nullable = false)
    private Long childId;

    @Column(nullable = false)
    private String name;

    private String relationship;

    @Column(name = "contact_number")
    private String contactNumber;

    private String email;

    private String address;

    @Column(name = "is_primary")
    private Boolean isPrimary = false;

    @Column(name = "user_id")
    private Long userId;

    public Guardian() {}

    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }

    public Long getChildId() { return childId; }
    public void setChildId(Long childId) { this.childId = childId; }

    public String getName() { return name; }
    public void setName(String name) { this.name = name; }

    public String getRelationship() { return relationship; }
    public void setRelationship(String relationship) { this.relationship = relationship; }

    public String getContactNumber() { return contactNumber; }
    public void setContactNumber(String contactNumber) { this.contactNumber = contactNumber; }

    public String getEmail() { return email; }
    public void setEmail(String email) { this.email = email; }

    public String getAddress() { return address; }
    public void setAddress(String address) { this.address = address; }

    public Boolean getIsPrimary() { return isPrimary; }
    public void setIsPrimary(Boolean isPrimary) { this.isPrimary = isPrimary; }

    public Long getUserId() { return userId; }
    public void setUserId(Long userId) { this.userId = userId; }
}

File: mobile/app/src/main/java/com/daycarelog/app/data/model/Guardian.kt
package com.daycarelog.app.data.model

data class Guardian(
    val id: Long? = null,
    val childId: Long = 0,
    val name: String = "",
    val relationship: String? = null,
    val contactNumber: String? = null,
    val email: String? = null,
    val address: String? = null,
    val isPrimary: Boolean = false,
    val userId: Long? = null,
)

data class GuardianRequest(
    val name: String,
    val relationship: String? = null,
    val contactNumber: String? = null,
    val address: String? = null,
    val isPrimary: Boolean = false,
    val createPortalAccount: Boolean = false,
    val email: String? = null,
)

data class CreateGuardianResponse(
    val guardian: Guardian,
    val tempPassword: String?,
)

data class GuardianAccountResponse(
    val userId: Long,
    val name: String?,
    val email: String?,
    val contactNumber: String?,
    val address: String?,
    val relationship: String?,
    val children: List<ChildSummary>,
)

data class ChildSummary(
    val id: Long,
    val firstName: String,
    val lastName: String,
)
